server/jetstream_api.go

```diff
@@ -2288,16 +2288,14 @@ func (s *Server) jsStreamLeaderStepDownRequest(sub *subscription, c *client, _ *
 		}
 	}
 
-	// Call actual stepdown. Do this in a Go routine.
-	go func() {
-		mset.setLeader(false)
-		// TODO (mh) eventually make sure all go routines exited and all channels are cleared
-		time.Sleep(250 * time.Millisecond)
-		node.StepDown(preferredLeader)
-
+	// Call actual stepdown.
+	err = node.StepDown(preferredLeader)
```

Comment:
No need to call `StepDown` in a go routine after a 250ms delay when stepping down stream/consumer.
This was introduced in https://github.com/nats-io/nats-server/pull/3079, to not process messages while stepping down. This is not needed anymore, as the RAFT logic itself will take care of that.